templates/components/engines/typescript/agent/chat.ts

```diff
@@ -13,7 +13,19 @@ export async function createChatEngine() {
   if (index) {
     tools.push(
       new QueryEngineTool({
-        queryEngine: index.asQueryEngine(),
+        queryEngine: index.asQueryEngine({
+          preFilters: {
+            filters: [
+              // TODO: add filter to only show documents that are public or that are private and
```

Comment:
@leehuwuj this is the biggest todo, the user sends the ids his private documents and we want to use these documents